src/main/java/gov/cms/mat/cql_elm_translation/utils/cql/parsing/Cql2ElmListener.java

```diff
@@ -556,18 +568,29 @@ private Element resolve(String identifier, CompiledLibrary library) {
           currentContext, def.getPath() + "-" + def.getVersion() + "|" + def.getLocalIdentifier());
       libraryAccessor = def;
       try {
-        parseChildLibraries(def);
-        libraries.add(
+        var parsedLibrary =
```

Comment:
Nice work here!

Reply:
thank you! noticed a few diffs between QDM translator and FHIR translator, so moved over them. This should be refactored to make it common logic between these two services.